src/app/zap-templates/zcl/data-model/chip/time-synchronization-cluster.xml

```diff
@@ -0,0 +1,87 @@
+    <item name="MicrosecondsGranularity" value="0x4"/>
+  </enum>
+
+  <enum name="TimeSourceEnum" type="ENUM8">
+    <item name="None" value="0x0"/>
+    <item name="Unknown" value="0x1"/>
+    <item name="Admin" value="0x2"/>
+    <item name="NodeTimeCluster" value="0x3"/>
+    <item name="NonFabricSntp" value="0x4"/>
+    <item name="NonFabricNtp" value="0x5"/>
+    <item name="FabricSntp" value="0x6"/>
+    <item name="FabricNtp" value="0x7"/>
+    <item name="MixedNtp" value="0x8"/>
+    <item name="NonFabricSntpNts" value="0x9"/>
+    <item name="NonFabricNtpNts" value="0xa"/>
+    <item name="FabricSntpNts" value="0xb"/>
+    <item name="FabricNtpNts" value="0xc"/>
+    <item name="MixedNtpNts" value="0xd"/>
+    <item name="CloudSource" value="0xe"/>
+    <item name="Ptp" value="0xf"/>
+    <item name="Gnss" value="0x10"/>
+  </enum>
+
+  <cluster>
+    <domain>General</domain>
+    <name>TimeSync</name>
+    <code>0x0038</code>
+    <define>TIMESYNC_CLUSTER</define>
+    <description>This cluster is used for time synchronization.</description>
+
+    <attribute side="server" code="0x0001" define="UTC_TIME" type="INT64U" writable="false" default="" optional="false">UTCTime</attribute>
+    <attribute side="server" code="0x0002" define="GRANULARITY" type="ENUM8" writable="false" default="NoTimeGranularity" optional="false">Granularity</attribute>
+    <attribute side="server" code="0x0003" define="TIME_SOURCE" type="ENUM8" writable="false" default="None" optional="false">TimeSource</attribute>
+    <attribute side="server" code="0x0004" define="TRUSTED_TIME_NODE_ID" type="NODE_ID" writable="true" default="" optional="false">TrustedTimeNodeId</attribute>
+    <attribute side="server" code="0x0005" define="DEFAULT_NTP" type="CHAR_STRING" writable="true" default="" length="128" optional="false">DefaultNtp</attribute>
+    <attribute side="server" code="0x0006" define="TIME_ZONE" type="ARRAY" entryType="TimeZoneType" writable="true" default="" length="2" optional="false">TimeZone</attribute>
+    <attribute side="server" code="0x0007" define="DST_OFFSET" type="ARRAY" entryType="DstOffsetType" writable="true" default="" length="20" optional="false">DstOffset</attribute>
+    <attribute side="server" code="0x0008" define="LOCAL_TIME" type="INT64U" writable="false" default="0" optional="false">LocalTime</attribute>
+    <attribute side="server" code="0x0009" define="TIME_ZONE_DATABASE" type="boolean" writable="false" default="false" optional="false">TimeZoneDatabase</attribute>
+    <attribute side="server" code="0x000a" define="NTP_SERVER_PORT" type="INT16U" writable="false" default="" optional="false">NtpServerPort</attribute>
+	    
+    <command source="server" code="0x00" name="SetUtcTime" optional="false">
+      <description>Requests UTCTime update.</description>
+      <arg name="UtcTime" type="INT64U"/>
+      <arg name="Granularity" type="ENUM8"/>
+      <arg name="TimeSource" type="ENUM8"/>
```

Comment:
And using the right enum types here?

Reply:
This one _should_ work, though, I think.  See the ApplyUpdateRequestResponse command in `src/app/zap-templates/zcl/data-model/chip/chip-ota.xml` for an example of this working.